src/aws-cpp-sdk-core/source/auth/SSOCredentialsProvider.cpp

```diff
@@ -28,15 +28,20 @@ using Aws::Utils::Threading::ReaderLockGuard;
 
 static const char SSO_CREDENTIALS_PROVIDER_LOG_TAG[] = "SSOCredentialsProvider";
 
-SSOCredentialsProvider::SSOCredentialsProvider() : m_profileToUse(GetConfigProfileName())
+SSOCredentialsProvider::SSOCredentialsProvider() : SSOCredentialsProvider(GetConfigProfileName(), nullptr)
```

Comment:
instead of passing nullpointer why not construct the default instance of the client configuration here. that way you dont need to do the check later. i.e.
```cpp
SSOCredentialsProvider::SSOCredentialsProvider(const Aws::String& profile) : SSOCredentialsProvider(profile, Aws::MakeShared<ClientConfiguration>(SSO_CREDENTIALS_PROVIDER_LOG_TAG))
{
    m_config->scheme = Aws::Http::Scheme::HTTPS;
    m_config->region = m_ssoRegion;
    AWS_LOGSTREAM_DEBUG(SSO_CREDENTIALS_PROVIDER_LOG_TAG, "Passing config to client for region: " << m_ssoRegion);
    Aws::Vector<Aws::String> retryableErrors;
    retryableErrors.push_back("TooManyRequestsException");
    m_config->retryStrategy = Aws::MakeShared<SpecifiedRetryableErrorsRetryStrategy>(SSO_CREDENTIALS_PROVIDER_LOG_TAG, retryableErrors, 3/*maxRetries*/);
}
```

this saves us from having to create a default client configuration each time `Reload`

Reply:
Done, in a slightly different way.